woocommerce/woocommerce
# [Performance]: Replace `classnames` usage with `clsx`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Replace `classnames` with `clsx` across WooCommerce, which is a [faster](https://github.com/lukeed/clsx/tree/master/bench) and lighter alternative.

This is aligned with the equivalent recent change made on Gutenberg on https://github.com/WordPress/gutenberg/pull/61138

Closes #47602



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

No visible changes should be observed to the end user.

1. Make sure all tests on this PR pass.
2. Do a quick smoketest with a few Woo blocks & confirm there are no visual regressions.
3. Do a quick smoketest by accessing WooCommerce > Home > Customize Your Store and confirm there are no visual regressions.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Replaced `classnames` package with the faster and smaller `clsx` package.

#### Comment 

</details>
